contracts/external/gmxV2/GmxV2MarketTokenPriceOracle.sol

```diff
@@ -0,0 +1,100 @@
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+*/
+
+pragma solidity ^0.8.9;
+
+import { IERC20Metadata } from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+import { IDolomiteMargin } from "../../protocol/interfaces/IDolomiteMargin.sol";
+import { IDolomitePriceOracle } from "../../protocol/interfaces/IDolomitePriceOracle.sol";
+import { IDolomiteStructs } from "../../protocol/interfaces/IDolomiteStructs.sol";
+import { IGmxV2IsolationModeVaultFactory } from "../interfaces/gmx/IGmxV2IsolationModeVaultFactory.sol";
+
+import { Require } from "../../protocol/lib/Require.sol";
+
+import { IGmxRegistryV2 } from "../interfaces/gmx/IGmxRegistryV2.sol";
+
+import "hardhat/console.sol";
+
+/**
+ * @title   GmxV2MarketTokenPriceOracle
+ * @author  Dolomite
+ *
+ * @notice  An implementation of the IDolomitePriceOracle interface that gets GMX's V2 Market token price in USD
+ */
+contract GmxV2MarketTokenPriceOracle is IDolomitePriceOracle {
+
+    // ============================ Constants ============================
+
+    bytes32 private constant _FILE = "GmxV2MarketTokenPriceOracle";
+    uint256 public constant ETH_USD_PRECISION = 1e18;
+    uint256 public constant FEE_PRECISION = 10_000;
+
+    // ============================ Public State Variables ============================
+
+    // @note Revisit naming conventions
+    address public immutable DGM_ETH_USD; // solhint-disable-line var-name-mixedcase
+    IGmxRegistryV2 public immutable REGISTRY; // solhint-disable-line var-name-mixedcase
+    IDolomiteMargin public immutable DOLOMITE_MARGIN; // solhint-disable-line var-name-mixedcase
+
+    // ============================ Constructor ============================
+
+    constructor(
+        address _dGmEthUsd,
+        address _gmxRegistryV2,
+        address _dolomiteMargin
+    ) {
+        DGM_ETH_USD = _dGmEthUsd;
+        REGISTRY = IGmxRegistryV2(_gmxRegistryV2);
+        DOLOMITE_MARGIN = IDolomiteMargin(_dolomiteMargin);
+    }
+
+    function getPrice(
+        address _token
+    )
+    public
+    view
+    returns (IDolomiteStructs.MonetaryPrice memory) {
+        Require.that(
+            _token == address(DGM_ETH_USD),
+            _FILE,
+            "Invalid token",
+            _token
+        );
+        Require.that(
+            DOLOMITE_MARGIN.getMarketIsClosing(DOLOMITE_MARGIN.getMarketIdByTokenAddress(_token)),
+            _FILE,
+            "gmEthUsd cannot be borrowable"
+        );
+
+        return IDolomiteStructs.MonetaryPrice({
+            value: _getCurrentPrice()
+        });
+    }
+
+    // ============================ Internal Functions ============================
+
+    // @note Look at stuff GMX sent
+    function _getCurrentPrice() internal view returns (uint256) {
+        IGmxV2IsolationModeVaultFactory factory = IGmxV2IsolationModeVaultFactory(DGM_ETH_USD);
+
+        console.log(factory.indexToken());
+        return 1;
```

Comment:
Make this file generic for any GM token, etc. etc. We talked about this over Slack/calls